### PR TITLE
Add tracking logs

### PR DESCRIPTION
**Pull request by pomegranited**
**Description:**

This PR is the first step for https://github.com/openedx/openedx-aspects/issues/222

It is a rebase of https://github.com/open-craft/openedx-completion-aggregator/pull/173 plus:
* emitted event prefix is `openedx` instead of `edx`
* removed `openedx.completion_aggregator.completion.*` events -- only emitting `progress`
* edx-completion upgraded to edx-platform version

Progress events will be emitted indicating any changes to aggregated progress for these block types:
```
    openedx.completion_aggregator.progress.course
    openedx.completion_aggregator.progress.chapter
    openedx.completion_aggregator.progress.sequential
    openedx.completion_aggregator.progress.vertical
```

**JIRA:** [FAL-3766](https://tasks.opencraft.com/browse/FAL-3766) (private link)

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** ASAP -- there's follow-up work to come soon.

**Installation instructions:**
1. Install this package in your LMS venv
1. Ensure completion is enabled : should be a waffle switch enabled for [completion.enable_completion_tracking](http://local.edly.io:8000/admin/waffle/switch/)
1. Create a course with one or two subsections, sections, blocks.

Testing instructions:

1. Watch the tracking.log file generated by your LMS for completion events, e.g `tail -f $(tutor config printroot)/data/lms/logs/tracking.log | grep completion_aggregator`
1. Enrol in your test course and complete some blocks.
   * You should see `openedx.completion_aggregator.progress.*` events coming through to mark the progress towards completion of the units (vertical), subsections (sequential), sections (chapter), and course.
   * When you've completed the children of these blocks, you should see corresponding openedx.completion_aggregator.completion.* events.

**Reviewers:**
- [x] @Agrendalath 

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [x] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [x] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Review by Ian2012 — APPROVED** [review]

**Review by Agrendalath — APPROVED** [review]
@pomegranited, before merging this, could you please add information about the tracking logs to the `README` along with a warning that this can generate plenty of additional events? We should also mention how to turn this off. Please see the "Synchronous vs. Asynchronous calculations" for an example format of this warning.

:+1:

* [x] I tested this: 
* [x] I read through the code
* [x] I checked for accessibility issues: n/a
* [ ] Includes documentation: ⚠️

File: completion_aggregator/__init__.py [hunk 1]
```diff
@@ -5,4 +5,4 @@
 
 from __future__ import absolute_import, unicode_literals
 
-__version__ = '4.0.3'
+__version__ = '4.0.4'
```
[the remark this answers is not in the document]
Comment (pomegranited) [reply completion_aggregator/__init__.py]: Done, see [change](https://github.com/open-craft/openedx-completion-aggregator/pull/206/files#diff-06149580981d6f396e99975bd45f6a345d34cad5bc810d6f03d57cb340310aeeR8).

File: test_settings.py [hunk 2]
```diff
@@ -18,10 +18,24 @@ def root(*args):
...
+        "chapter",
+        "sequential",
+        "vertical",
+    ],
+    "completion": [
+        "course",
+        "chapter",
+        "sequential",
+        "vertical",
+    ]
```
[the remark this answers is not in the document]
Comment (pomegranited) [reply test_settings.py]: Fixed, see [change](https://github.com/open-craft/openedx-completion-aggregator/pull/206/files#diff-01c96a2a4d7b08430652ab4c332ad94f6c35db72082b7b2a6f26077bf9b83696R21-R34).

File: completion_aggregator/models.py [hunk 3]
```diff
@@ -171,8 +172,44 @@ def submit_completion(self, user, course_key, block_key, aggregation_name, earne
...
+    def emit_completion_aggregator_logs(updated_aggregators):
+        """
+        Emit a tracking log for each element of the list parameter.
+
+        Parameters
+        ----------
+            updated_aggregators: List of Aggregator intances
+
+        """
+        for aggregator in updated_aggregators:
```
[the remark this answers is not in the document]
Comment (pomegranited) [reply completion_aggregator/models.py:192]: Sure -- this might help speed things up a little when batch processing where events aren't wanted.

**Comment by pomegranited** [comment]
Thank you for your review @Agrendalath !

> before merging this, could you please add information about the tracking logs to the README along with a warning that this can generate plenty of additional events? We should also mention how to turn this off.

Good point -- I [added a section to the README](https://github.com/open-craft/openedx-completion-aggregator/pull/206/files#diff-7b3ed02bc73dc06b7db906cf97aa91dec2b2eb21f2d92bc5caa761df5bbc168fR50-R58) -- is that clear enough?

File: test_settings.py [hunk 4]
```diff
@@ -18,10 +18,24 @@ def root(*args):
...
+        "sequential",
+        "vertical",
+    },
+    "completion": {
+        "course",
+        "chapter",
+        "sequential",
+        "vertical",
+    }
+}
```
Comment (pomegranited) [inline test_settings.py:34]: Hi @Agrendalath 

I'm reconsidering these event types.. we're only emitting a "progressed" event for [ordinary completions](https://github.com/openedx/event-routing-backends/blob/master/event_routing_backends/processors/xapi/event_transformers/completion_events.py#L13), with the `completed` flag indicating completion. I'm going to do the same here, and remove these "completion" event types.

**Review by Agrendalath — APPROVED** [review]
@pomegranited, the changes look good. I just left some minor suggestions.

**Comment by pomegranited** [comment]
Thank you @Agrendalath :) I've addressed your comments, thank you! Will merge and tag this change.

**Pull request merged**